packages/components/src/components/Table/TableCell.tsx

```diff
@@ -1,26 +1,55 @@
-import { spacingsPx, typography } from '@trezor/theme';
 import { ReactNode } from 'react';
-import styled, { css } from 'styled-components';
+import styled from 'styled-components';
 
-export interface TableCellProps {
-    children: ReactNode;
-    $alignRight?: boolean;
-    $isHeader?: boolean;
-}
+import { typography } from '@trezor/theme';
 
-export const TableCell = styled(({ $isHeader, $alignRight, ...props }: TableCellProps) =>
-    $isHeader ? <th {...props} /> : <td {...props} />,
-)<TableCellProps>`
+import { FrameProps, FramePropsKeys, withFrameProps } from '../../utils/frameProps';
+import { makePropsTransient, TransientProps } from '../../utils/transientProps';
+
+export const allowedTableCellFrameProps: FramePropsKeys[] = ['margin'];
+type AllowedFrameProps = Pick<FrameProps, (typeof allowedTableCellFrameProps)[number]>;
+
+export type TableCellProps = AllowedFrameProps & {
+    children?: ReactNode;
+    align?: 'left' | 'right';
+    isHeader?: boolean;
+    isSmaller?: boolean;
```

Comment:
Using this solution, the icon column has different size in legit and hidden tokens (different number of columns). It does not look nice.

And instead of `isSmaller`, I would like something more generic. I would like to be able to set `grid-template-columns` to the table or `grid-column` to cells. I don't know what is best solution, but this is not generic.

Please help @jvaclavik @adamhavel

Reply:
I quite agree with @tomasklim. Setting `grid-template-columns` or `grid-column` is much more universal than this. 

let's wait for @adamhavel 's opinion, he is the grid guru 🙂

Reply:
Well I suggest to keep it as simple as possible. I'm thinking props like columns={n} on the Table and span={n} on the TableCell. But if we wanna set grid on the Table, then we'd have to use subgrid on the rows. It's doable but maybe we could just use native table rendering and not override it. Then we could use attributes like colspan. By the way, I'm creating a simple Grid implementation in the design system right now, but I'd like to keep it separate from the Table (semantics and so on)